test/test_irods_http_api.py

```diff
@@ -1825,6 +1825,332 @@ def test_non_parallel_writes_with_data_exceeding_internal_write_threshold(self):
+            # have a parallel-write handle.
+            invoke_parallel_write_shutdown = True
+
+            # Write to the data object using the parallel write handle.
+            futures = []
+            with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
+                for e in enumerate(['A', 'B', 'C']):
+                    count = 10
+                    futures.append(executor.submit(self.execute_write_using_header_based_form, **{
+                        'bearer_token': self.rodsuser_bearer_token,
+                        'fields': {
+                            'op': 'write',
+                            'parallel-write-handle': parallel_write_handle,
+                            'offset': e[0] * count,
+                            'stream-index': e[0]
+                        },
+                        'bytes': e[1] * count
+                    }))
+
+                for f in concurrent.futures.as_completed(futures):
+                    result = f.result()
+                    self.logger.debug(result)
+                    self.assertEqual(result['irods_response']['status_code'], 0)
+
+            # End the parallel write.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'parallel_write_shutdown',
+                'parallel-write-handle': parallel_write_handle
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            # Read the contents of the data object and show it contains exactly what we expect.
+            r = requests.get(self.url_endpoint, headers=headers, params={
+                'op': 'read',
+                'lpath': data_object,
+                'count': 30
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.content.decode('utf-8'), 'A' * 10 + 'B' * 10 + 'C' * 10)
+
+        finally:
+            # End the parallel write in case something failed.
+            # This avoids leaking of parallel-write resources in the HTTP API server.
+            if invoke_parallel_write_shutdown:
+                r = requests.post(self.url_endpoint, headers=headers, data={
+                    'op': 'parallel_write_shutdown',
+                    'parallel-write-handle': parallel_write_handle
+                })
+                self.logger.debug(r.content)
+
+            # Remove the data object.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'catalog-only': 0,
+                'no-trash': 1
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
```

Comment:
I noticed that `test_writing_to_data_object_using_header_based_form` and the other new tests did not make assertions after removing its data object. Not saying this is right or wrong, just pointing out that this one is different.

Reply:
These assertions shouldn't be there. The reason there's a difference is due to the age of tests. I believe the tests which do not contain these assertions are newer than the one which I derived this test from.

Reply:
Removed assertions.